### PR TITLE
feat: simple release signing system

### PR DESCRIPTION
**Pull request by dpc**
Use `just sign-release v0.2.2` to sign the release.

`<prefix>-SHA256SUM` and `<prefix>-SHA256SUMS.asc` are meant to be checked in.

Now we get to (hopefully :crossed_fingers:) rip the benefits of Nix. We don't need the CI to sign anything, and anyone can verify and sign the release independently from the comfort of their personal terminal.

For reviewers:

* please try to sign and verify that checksum are deterministic, you don't have to submit your signatures
* does the file naming and fs structure make sense?

We can iterate from here

Immediate follow-ups:

* [ ] add and verify reproducibility of macos binaries
* [ ] check-in public keys from maintainers that are supposed to sign releases

### CODE REVIEW

File: scripts/release/sign.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+fi
+
+tag="$1"
+system="$(nix eval --raw --impure --expr builtins.currentSystem)"
+prefix="${tag}-${system}"
+release_dir_base="releases/bins"
+release_dir="${release_dir_base}/${prefix}"
+sha256sum_path="releases/${prefix}.SHA256SUMS"
+
+mkdir -p "${release_dir}"
```
Comment (elsirion) [inline scripts/release/sign.sh:16]: The `bins` path should likely be in `.gitignore`

File: releases/v0.2.2-x86_64-linux.SHA256SUMS [hunk 2]
```diff
@@ -0,0 +1,6 @@
+716840cfe02a270aff5bf1a02eda9c87946af231ad4ca5796a981ed0d9b1723f  v0.2.2-x86_64-linux/fedimint-cli
```
Comment (elsirion) [inline releases/v0.2.2-x86_64-linux.SHA256SUMS:1]: I'm getting the same hashes

**Review by elsirion — APPROVED** [review]
:tada:

File: releases/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+!/bins
```
Comment (elsirion) [inline releases/.gitignore:1]: Somehow this doesn't work for me? I'm getting

```
$ git status 
HEAD detached at pr/4339/head
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   v0.2.2-x86_64-linux.SHA256SUMS.asc

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	bins/

no changes added to commit (use "git add" and/or "git commit -a")
```

File: releases/v0.2.2-x86_64-linux.SHA256SUMS (hunk 2 above)
Comment (m1sterc001guy) [reply releases/v0.2.2-x86_64-linux.SHA256SUMS:1]: image

Same as me :)

File: releases/.gitignore (hunk 3 above)
Comment (m1sterc001guy) [reply releases/.gitignore:1]: My `git status` has the bins under `releases/bins`, which is still showing up untracked in git

**Review by m1sterc001guy — APPROVED** [review]

File: releases/.gitignore (hunk 3 above)
Comment (dpc) [reply releases/.gitignore:1]: https://github.com/fedimint/fedimint/pull/4367

File: scripts/release/sign.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+
+>&2 echo "Building..."
+for out in fedimint-pkgs gateway-pkgs ; do
+  nix build "git+file:.?ref=refs/tags/${tag}#${out}"
```
Comment (elsirion) [inline scripts/release/sign.sh:21]: I was wondering why we aren't using this build command in the context of #4305, but would these binaries run on non-nix systems? I doubt it and that's why we use bundlers. We should probably rather sign the bundler-generated binaries that we actually publish.

Comment (dpc) [reply scripts/release/sign.sh:21]: Unclear if these binaries are reproducible but let's try. :D

Comment (dpc) [reply scripts/release/sign.sh:21]: https://github.com/fedimint/fedimint/pull/4372

**Pull request merged**